test/controllers/graphql_controller_12_test.rb

```diff
@@ -680,4 +680,55 @@ def teardown
   assert_response :success
   assert_equal 'science', JSON.parse(@response.body)['data']['createProjectMedia']['project_media']['tags']['edges'][0]['node']['tag_text']
   end
+
+  test "should not create duplicate tags for the ProjectMedia and FactCheck" do
+    Sidekiq::Testing.inline!
+    t = create_team
+    a = ApiKey.create!
+    b = create_bot_user api_key_id: a.id
+    create_team_user team: t, user: b
+    p = create_project team: t
+    authenticate_with_token(a)
+
+    query1 = ' mutation create {
+              createProjectMedia(input: {
+                project_id: ' + p.id.to_s + ',
+                media_type: "Blank",
+                channel: { main: 1 },
+                set_tags: ["science", "science", "#science"],
```

Comment:
also set_tags: ["science", "science", "#science", 'SCIENCE', '#SCIENCE'] should pass